### PR TITLE
[pack] Enhancements to JSON output to support files

### PR DESCRIPTION
**Pull request by fabiocav**
@ankitkumarr still iterating on a couple of things, but wanted to have this out for initial feedback and to get the artifacts.

### CODE REVIEW

File: src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs [hunk 1]
```diff
@@ -88,11 +121,24 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
-                ColoredConsole.WriteLine($"[{DateTime.UtcNow.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffZ", CultureInfo.InvariantCulture)}] ".DarkGray() + line);
+                string prefix = includeTimeStamp
+                    ? $"[{DateTime.UtcNow.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffZ", CultureInfo.InvariantCulture)}] ".DarkGray().ToString()
+                    : string.Empty;
+
+                ColoredConsole.WriteLine(prefix + line);
+            }
+        }
+
+        private void LogJsonToFile(string message)
```
Comment (ankitkumarr) [inline src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs:136]: To be fully sure we are on the same page, the file would not be a valid JSON but rather contain JSON every line?
Example --
```
{ "a":"b"}
{"d":"e"}
```

Comment (fabiocav) [reply src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs:136]: That is correct. See https://jsonlines.org/ for some examples

File: src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs [hunk 2]
```diff
@@ -74,12 +101,18 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
+                return;
+            }
+            
             if (DoesMessageStartsWithAllowedLogsPrefix(formattedMessage))
             {
                 LogToConsole(logLevel, exception, formattedMessage);
                 return;
             }
-
+
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs:115]: this code actually needs to be removed

File: src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs [hunk 3]
```diff
@@ -263,6 +270,14 @@ private async Task<IWebHost> BuildWebHost(ScriptApplicationHostOptions hostOptio
             if (EnableJsonOutput != null && EnableJsonOutput.Value)
             {
                 Environment.SetEnvironmentVariable("FUNCTIONS_ENABLE_JSON_OUTPUT", bool.TrueString);
+
+                if (JsonOutputFile != null)
+                {
+                    if (File.Exists(JsonOutputFile))
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs:276]: this code actually needs to be removed

**Comment by fabiocav** [comment]
Updated

**Review by ankitkumarr — APPROVED** [review]

**Comment by fabiocav** [comment]
Build was flaky (unrelated tests), but now green. Merging.

**Pull request merged**